faster_whisper/transcribe.py

```diff
@@ -1167,6 +1941,408 @@ def find_alignment(
             )
         ]
 
+    def encode_batch(self, features: torch.Tensor) -> ctranslate2.StorageView:
+        # When the model is running on multiple GPUs, the encoder output should be moved
+        # to the CPU since we don't know which GPU will handle the next job.
+        to_cpu = self.model.device == "cuda" and len(self.model.device_index) > 1
+        if features.device.type != "cpu":  # for GPU pipeline iterator
+            features = features.cpu().numpy()
+        features = get_ctranslate2_storage(features)
+        return self.model.encode(features, to_cpu=to_cpu)
+
+    def assign_word_timings(self, alignments, text_token_probs, words, word_tokens):
+        text_indices = np.array([pair[0] for pair in alignments])
+        time_indices = np.array([pair[1] for pair in alignments])
+
+        if len(word_tokens) <= 1:
+            return []
+
+        word_boundaries = np.pad(np.cumsum([len(t) for t in word_tokens[:-1]]), (1, 0))
+        if len(word_boundaries) <= 1:
+            return []
+
+        jumps = np.pad(np.diff(text_indices), (1, 0), constant_values=1).astype(bool)
+        jump_times = time_indices[jumps] * TIME_PRECISION
```

Comment:
Should calculator from `self.feature_extractor`:
```python
tokens_per_second = self.feature_extractor.sampling_rate // (
    self.feature_extractor.hop_length * self.input_stride
)
jump_times = time_indices[jumps] / tokens_per_second
```

Reply:
It was already there, updated now with `self.tokens_per_second`